src/subcommands/common.rs

```diff
@@ -26,12 +26,13 @@ impl ProviderOpts {
     /// The Core Provider cannot be used and will be overriden by the
     /// service default.
     pub fn provider(&self) -> Result<ProviderID, ParsecToolError> {
-        if self.provider != 0 {
-            Ok(ProviderID::try_from(self.provider)?)
-        } else {
-            let mut client = BasicClient::new_naked();
-            client.set_default_provider()?;
-            Ok(client.implicit_provider())
+        match self.provider {
+            None => {
+                let mut client = BasicClient::new_naked();
+                client.set_default_provider()?;
```

Comment:
I'm wondering if we should make the `set_default_...` methods return the default at the end.

Reply:
Yeah would be a good addition